src/foam/u2/view/LoginView.js

```diff
@@ -0,0 +1,309 @@
+  .horizontal-flip {
+    -moz-transform: scale(-1, 1);
+    -webkit-transform: scale(-1, 1);
+    -o-transform: scale(-1, 1);
+    -ms-transform: scale(-1, 1);
+    transform: scale(-1, 1);
+    margin-right: 10px;
+  }
+  /* *** ON SMESTYLES ??? */
+  .inline-block {
+    display: inline-block;
+  }
+
+/* ON LEFT SIDE IMG */
+  ^ .cover-img-block1 {
+    position: sticky;
+  }
+  ^ .image-one {
+    width: 45vw;
+    margin-top: -90vh;
+    margin-left: 5em;
+  }
+  `,
+
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'topBarShow_',
+      factory: function() {
+        if ( this.appConfig && this.appConfig.url ) {
+          return true;
+        }
+        return false;
+      },
+      hidden: true
+    },
+    {
+      name: 'model',
+      factory: () => {
+       return {};
+      },
+      view: { class: 'foam.u2.detail.VerticalDetailView' }
+    },
+    'param',
+    {
+      class: 'String',
+      name: 'mode_',
+      hidden: true
+    },
+    {
+      class: 'String',
+      name: 'imgPath',
+      expression: function(loginVariables) {
+        return loginVariables.imgPath || '';
+      }
+    },
+    {
+      class: 'String',
+      name: 'backLinkTxt_',
+      factory: function() {
+        let temp = this.backLink_.includes('www.') ?
+          this.backLink_.substring(this.backLink_.indexOf('www.') + 4) :
+          this.backLink_;
+        return temp.includes('://') ?
+          temp.substring(temp.indexOf('://') + 3) :
+          temp;
+      }
+    },
+    {
+      class: 'String',
+      name: 'backLink_',
+      factory: function() {
+        return this.model.backLink_ ? this.model.backLink_ : this.appConfig.url;
+      },
+      hidden: true
+    }
+  ],
+
+  messages: [
+    { name: 'GO_BACK', message: 'Go to ' },
+    { name: 'MODE1', message: 'SignUp' }
+  ],
+
+  methods: [
+    function init() {
+      if ( ! this.param ) {
+        this.param = {};
+      }
+      this.param.dao_ = ! this.param.dao_ ? this.loginVariables.dao_ : this.param.dao_;
+      if ( this.mode_ === this.MODE1 ) {
+        this.param.group_ = ! this.param.group_ ? this.loginVariables.group_ : this.param.group_;
+        this.param.countryChoices_ = this.param.countryChoices_ ? this.param.countryChoices_ : ['CA', 'US'];
+        this.model = this.SignUp.create(this.param, this);
+      } else {
+        this.model = this.SignIn.create(this.param, this);
+      }
+    },
+    function initE() {
+      this.SUPER();
+      // CREATE MODEL VIEW
+      var right = this.Element.create({}, this)
+      // Header on-top of rendering model
+        .start().show(this.topBarShow).addClass('topBar-logo-Back')
+          .start('img')
+            .attr('src', this.theme.logo)
+            .addClass('top-bar-img')
+          .end()
+      .end()
+      // Title txt and Model
+        .start().addClass('title-top').add(this.model.TITLE).end()
+        .startContext({ data: this })
+          .addClass('content-form').tag(this.MODEL).br()
```

Comment:
Why not make this into a slot so we don't have to push new loginView to update the page? It would make this view a lot more dynamic. I'd like to see that logic applied with the rest of this initE.

Reply:
Not sure what you are implying here? this.model is a slot, and if we wanted to not call LoginView, then how would that work? I see each view as a page, calling LoginView for two separate scenarios and for two seperate pages. If we don't call LoginView then  ... ???

This does bring a point though, in the footerLink(), I wouldn't have to call LoginView, if I was aware of the calling view in the model ... food for thought... will consider this. But otherwise, don't see away to avoid calling LoginView.

Reply:
Disregard my comment. I'm was thinking we'd have a constant reference to a loginView on the applicationController but that's not the best idea.